src/main/java/com/iexec/worker/executor/TaskNotificationService.java

```diff
@@ -52,6 +54,10 @@ public class TaskNotificationService {
     private final ContributionService contributionService;
     private final IexecHubService iexecHubService;
 
+    private final Map<String, Long> finalDeadlineForTask = ExpiringMap
+            .builder()
+            .expiration(1, TimeUnit.HOURS)
```

Comment:
Why 1 hour instead of 10T?

Reply:
How do you define `10T` in this context? We work with multiple categories. I had to use a arbitrary value. We may want to increase the timeout but we have to be aware of not storing too much things for too long. Thus, 1 hour seemed to be a not-so-bad value (and this is the value we kept for other devs so it is consistent).

Reply:
I was thinking about reading in constructor the biggest category through iexec hub service.
Anyway for today 1 hour today is fine, thank you :)